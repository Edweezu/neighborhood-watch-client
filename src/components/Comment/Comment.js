import React from 'react'
import MainContext from '../../contexts/MainContext'
import 'react-confirm-alert/src/react-confirm-alert.css';
import moment from 'moment'
import EditCommentModal from '../EditCommentModal/EditCommentModal'
import PropTypes from 'prop-types'


class Comment extends React.Component {

    static contextType = MainContext    
    static defaultProps = {
       user: {},
       nameCapitalized: () => {
       }
    }
    
    dateDiff = () => {
        let { date_created } = this.props
        let niceDate = moment(date_created, 'ddd MMM DD YYYY HH:mm:ss ZZ').format("MMM DD, YYYY")

        //how to change from provided server UTC time to local time so calculations are correct
        let daysAgo = moment.utc(date_created).local().fromNow()
        let todayUnix = moment().unix()
        let dateUnix = moment(new Date(date_created)).unix()

        //need to divide unix time by 86400 seconds in a day to find day diff
        let dateDiff = (todayUnix - dateUnix)/86400

        if (dateDiff > 6) {
            return `Posted on ${niceDate}`
        } else {
            return `Posted ${daysAgo}`
        }
    } 


    render () {
        const { id, user, text, nameCapitalized, deleteNumComment } = this.props

        return (
            <section className='Comment'>
                <div className='Comment__userInfo'>
                    {nameCapitalized(user.username)}, {user.city}
                </div>
                <div className='Post__body'>
                    {text}
                </div>
                <div className='Comment__date'>
                    <p>{this.dateDiff()}</p>
                    <EditCommentModal
                        commentId={id}
                        deleteNumComment={deleteNumComment}
                    />
                </div>  
            </section>
        )
    }
}

Comment.propTypes = {
    nameCapitalized: PropTypes.func
}

export default Comment