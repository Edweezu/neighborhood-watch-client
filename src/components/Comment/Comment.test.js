import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Comment from './Comment'


describe('Comment component', () => {
    const props = {
        id: 1,
        user: {
            username: 'demo',
            city: 'Santa Monica'
        },
        text: 'random text',
        nameCapitalized: () => {
            
        }
    }

    it('renders the given component', () => {
        const wrapper = shallow(<Comment />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<Comment {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})