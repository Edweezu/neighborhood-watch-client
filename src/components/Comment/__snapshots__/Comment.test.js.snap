// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component renders the component's given props 1`] = `
<section
  className="Comment"
>
  <p>
    , 
    Santa Monica
  </p>
  <p>
    random text
  </p>
  <p>
    Posted a few seconds ago
  </p>
  <EditCommentModal
    commentId={1}
  />
</section>
`;

exports[`Comment component renders the given component 1`] = `
<section
  className="Comment"
>
  <p>
    , 
  </p>
  <p />
  <p>
    Posted a few seconds ago
  </p>
  <EditCommentModal />
</section>
`;
